overlord/snapstate/snapstate.go

```diff
@@ -95,8 +95,15 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 		return nil, err
 	}
 
-	// ensure core gets installed. if it is already installed return
-	// an empty task set
+	if snapst.IsInstalled() {
```

Comment:
why not `if snapsup.PlugsOnly && snapst.IsInstalled()` (and then just `snapsup.PlugsOnly = len(info.Slots) == 0` ?